src/external.py

```diff
@@ -0,0 +1,188 @@
+        )
+        self._add_typeparam(params)
+
+    def _attach_hook(self):
+        sim = self._simulation
+
+        cpp_class = getattr(_azplugins, self._make_cpp_class_name())
+
+        if self.geometry not in ("film", "droplet"):
+            raise ValueError(f"Unrecognized geometry: {self.geometry}")
+
+        self._cpp_obj = cpp_class(sim.state._cpp_sys_def, self.interface)
+
+        super()._attach_hook()
+
+    def _make_cpp_class_name(self):
+        cpp_class_name = self.__class__.__name__
+        if isinstance(self._simulation.device, hoomd.device.GPU):
+            cpp_class_name += "GPU"
+        return cpp_class_name
+
+
+class PlanarHarmonicBarrier(HarmonicBarrier):
+    r"""Applies a purely repulsive harmonic potential to particles
+        near an interface in planar geometry..
+
+    Args:
+        interface (`hoomd.variant`) : `z` position of the interface
+
+    `PlanarHarmonicBarrier` apply a purely harmonic potential in planar geometry
+    that pushes down on nonvolatile solutes. The potential is truncated at its minimum
+    and have following form:
+
+    .. math::
+
+        \begin{eqnarray*}
+        V(d) = & 0 & d < H \\
+               & \frac{\kappa}{2} (d-H)^2 & H < d
+        \end{eqnarray*}
+
+    Here, the interface is located at height *H*, and may change with time.
+    The effective interface position *H* may be modified per-particle-type
+    using a *offset* (*offset* is added to *H* to determine the effective *H*)
+    :math:`\kappa` is a spring constant setting the strength of the interface
+    (:math:`\kappa` is a proxy for the surface tension).
+
+    Example::
+
+        interf = hoomd.variant.Ramp(A=50.0, B=25.0, t_start=100, t_ramp=1e6)
+        evap = hoomd.azplugins.external.PlanarHarmonicBarrier(interface = interf)
+
+        evap.params['S'] = dict(k=50.0, offset=0.0)
+        evap.params['B'] = dict(k=50.0*2*2, offset=0.0)
+
+    .. py:attribute:: params
+
+        The `PlanarHarmonicBarrier` parameters. The dictionary has the following
+        keys:
+
+        * ``k`` (`float`, **required**) - Spring constant :math:`k`
+          :math:`[\mathrm{energy} \cdot \mathrm{length}^{-2}]`
+        * ``offset`` (`float`, **required**) - Shift amount per-particle-type :math:`H`
+          :math:`[\mathrm{length}]`
+
+
+        Type: :class:`~hoomd.data.typeparam.TypeParameter` [`tuple`
+        [``particle_type``], `dict`]
+
+    .. warning::
+        Virial calculation has not been implemented for this model because it is
+        nonequilibrium. A warning will be raised if any calls to
+        :py:class:`hoomd.logging.Logger` are made because the logger always requests
+        the virial flags. However, this warning can be safely ignored if the
+        pressure (tensor) is not being logged or the pressure is not of interest.
+
+    """
+
+    def __init__(self, interface=None):
+        super().__init__(interface=interface, geometry="film")
```

Comment:
You are just going to call the parent constructor after you make the fixes above, so this can just be pass.
```suggestion
    pass
```